circuits/cpp/src/aztec3/circuits/kernel/public/native_public_kernel_circuit_no_previous_kernel.cpp

```diff
@@ -0,0 +1,58 @@
+#include "init.hpp"
+
+#include <aztec3/circuits/abis/public_kernel/public_kernel_inputs_no_previous_kernel.hpp>
+#include <aztec3/circuits/abis/kernel_circuit_public_inputs.hpp>
+#include "native_public_kernel_circuit_no_previous_kernel.hpp"
+#include "common.hpp"
+
+#include <aztec3/utils/array.hpp>
+#include <aztec3/utils/dummy_composer.hpp>
+#include <aztec3/circuits/hash.hpp>
+#include "aztec3/constants.hpp"
+
+namespace {
+void initialise_end_values(PublicKernelInputsNoPreviousKernel<NT> const& public_kernel_inputs,
+                           KernelCircuitPublicInputs<NT>& circuit_outputs)
+{
+    circuit_outputs.constants.historic_tree_roots.public_data_tree_root =
+        public_kernel_inputs.public_call.public_data_tree_root;
+    circuit_outputs.constants.tx_context = public_kernel_inputs.signed_tx_request.tx_request.tx_context;
+}
+} // namespace
+
+namespace aztec3::circuits::kernel::public_kernel {
+
+using aztec3::circuits::abis::KernelCircuitPublicInputs;
+using aztec3::circuits::abis::public_kernel::PublicKernelInputsNoPreviousKernel;
+using aztec3::circuits::kernel::public_kernel::common_validate_inputs;
+using aztec3::circuits::kernel::public_kernel::common_validate_kernel_execution;
+using aztec3::circuits::kernel::public_kernel::validate_function_execution;
+using aztec3::circuits::kernel::public_kernel::validate_state_reads;
+using aztec3::circuits::kernel::public_kernel::validate_state_transitions;
+using aztec3::circuits::kernel::public_kernel::validate_this_public_call_stack;
+using aztec3::utils::push_array_to_array;
+
+using DummyComposer = aztec3::utils::DummyComposer;
+
+KernelCircuitPublicInputs<NT> native_public_kernel_circuit_no_previous_kernel(
+    DummyComposer& composer, PublicKernelInputsNoPreviousKernel<NT> const& public_kernel_inputs)
+{
+    // There is not circuit state carried over from previous iterations.
+    // We are construcitng fresh state that will be added to during this circuit execution.
+    KernelCircuitPublicInputs<NT> public_inputs{};
+
+    // initialise the circuit end state with defaults and constants from the provided input
+    initialise_end_values(public_kernel_inputs, public_inputs);
+
+    // validate the inputs common to all invocation circumstances
+    common_validate_inputs(composer, public_kernel_inputs);
+
+    // validate the kernel execution commonn to all invocation circumstances
+    common_validate_kernel_execution(composer, public_kernel_inputs);
+
```

Comment:
We should add a TODO (or an issue) to remind us that we're not-yet checking existence of the public function in the contract tree. 
This is because we need to chat a bit more about what data is actually stored in the contract tree for a public function, and therefore what data we can actually use for the membership check.